pkgs/development/ocaml-modules/hmap/default.nix

```diff
@@ -0,0 +1,37 @@
+{ lib
+, buildOcaml
+, fetchurl
+, topkg
+}:
+
+buildOcaml rec {
+	name = "hmap-${version}";
+  version = "0.8.1";
+
+  minimumSupportedOcamlVersion = "4.02.0";
+    
+  src = fetchurl {
+    url = "http://erratique.ch/software/hmap/releases/${name}.tbz";
+    sha256 = "10xyjy4ab87z7jnghy0wnla9wrmazgyhdwhr4hdmxxdn28dxn03a";
+  };
+
+	buildInputs = [ topkg ];
+
+  inherit (topkg) buildPhase installPhase;
+
+  doCheck = true;
+
+  checkPhase = ''
+    runHook preCheck
+    ${topkg.run} build --tests true
```

Comment:
IIUC, this line belongs to the `buildPhase`

Reply:
Depends if you want buildPhase building both code and tests (not running tests), or just building code (and leaving the tests to be built in checkPhase). I see other OCaml packages doing what you suggest, so I'll follow that convention.